### PR TITLE
Add basic metrics to user notifications

### PR DESCRIPTION
**Pull request by artemruts**
### Description

This PR adds basic metrics to user notifications, specifically:
- when a user opens a notifications dropdown
- when a user clicks on the notification link

### User notifications are loaded, tracking cases:

**event name**: `UserNotificationsLoaded`
**payload**:  see `UserNotificationsOpened`, payload variations are the same

### User opens notifications dropdown, tracking cases:

1. **No notifications, everything is synced and up-to-date**

      **event name**: `UserNotificationsOpened`
      **payload**: 
      ```json
      {
        "status": "success"
      }
      ```
    <details>
     <summary>Everything is up-to-date demo</summary>
     <br/>
    All OK
    </details>

1. **No added code hosts** 

      **event name**: `UserNotificationsOpened`
      **payload**: 
      ```json
      {
        "status": "NoCodehosts"
      }
      ```
    <details>
     <summary>No code hosts demo</summary>
     <br/>
    No Codehosts
    </details>

1. **No added repositories**

      **event name**: `UserNotificationsOpened`
      **payload**: 
      ```json
      {
        "status": "NoRepos"
      }
      ```
    <details>
     <summary>No repos demo</summary>
     <br/>
    No Repos
    </details>
    
  1. **Multiple in-progress and error notifications**

      **event name**: `UserNotificationsOpened`
      **payload**: 
      ```ts
      {
        "status":  ["CloningProgress" | "IndexingProgress" | "ExternalServiceSyncError" | "SyncError" | "IndexingError"] 
      }
      ```
      <details>
       <summary>multiple notifications demo</summary>
       <br/>
      Multiple notification statuses
      </details>

### User clicks a notifications link:

**event name**: `UserNotificationsLinkClicked`
**payload**: 
```ts
{
  "href": string,
  "notificationType":  "not-active" | "progress" | "warning" | "success" | "error"
}
```
<details>
 <summary>notifications link tracking demo</summary>
 <br/>
 
 https://user-images.githubusercontent.com/1319181/135966649-78ccfc6c-b4c9-4d5e-a0a0-2e0b53485111.mov
</details>

## Tests

Tested locally and on .com

### CODE REVIEW

**Review by kopancek — APPROVED** [review]
LGTM

File: client/web/src/nav/StatusMessagesNavItem.tsx [hunk 1]
```diff
@@ -116,43 +117,56 @@ const getMessageColor = (entryType: EntryType): string => {
...
-                    {props.linkText}
-                </Link>
-            </div>
-        )}
-        {props.progressHint && <small className="text-muted">{props.progressHint}</small>}
-    </div>
-)
+const StatusMessagesNavItemEntry: React.FunctionComponent<StatusMessageEntryProps> = props => {
+    const onLinkClick = (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>): void => {
+        const payload = { href: props.linkTo, notificationType: props.entryType }
```
Comment (attfarhan) [inline client/web/src/nav/StatusMessagesNavItem.tsx:122]: What are the possible values of `href`? Want to make sure we are not allowing things like repo names / search queries with private code to get leaked here

**Review by attfarhan — APPROVED** [review]
I have one comment to be addressed, but if it's resolved/not a concern, approved 👍🏾

File: client/web/src/nav/StatusMessagesNavItem.tsx (hunk 1 above)
Comment (artemruts) [reply client/web/src/nav/StatusMessagesNavItem.tsx:122]: Good point, I'll remove this or see if I can figure out what we can track!

**Review by ryphil — APPROVED** [review]

**Pull request merged**